src/ui/components/Library/Team/View/TestRuns/TestRunList.tsx

```diff
@@ -1,20 +1,45 @@
-import { useContext } from "react";
+import { useContext, useState } from "react";
 
 import { LibrarySpinner } from "ui/components/Library/LibrarySpinner";
 import { TestRunListItem } from "ui/components/Library/Team/View/TestRuns/TestRunListItem";
+import { SecondaryButton } from "ui/components/shared/Button";
 
 import { TestRunsContext } from "./TestRunsContextRoot";
 
+const INITIAL_TEST_RUNS_DISPLAYED = 50;
```

Comment:
:question: Two thoughts:

- Can we limit the fetched data?
- Should this be a virtual list of some kind?

Reply:
Both good follow ups